py3status/modules/window_manipulation.py

```diff
@@ -0,0 +1,477 @@
+}
+
+# show names excluding floating and scratchpad
+window_manipulation {
+    format_window = '\?max_length=30 '
+    format_window += '[\?if=is_urgent [\?color=urgent&show \u2756] {name}'
+    format_window += '|[[\?if=is_focused [\?color=focused&show \u2756] {name}]'
+    format_window += '|[\?if=is_scratchpad '
+    format_window += '|[\?if=is_floating '
+    format_window += '|[[[\?color=unfocused&show \u2756] {name}]]]]]]'
+}
+
+# show almost everything but the magic of the customizations...
+# don't forget to add colors and buttons
+window_manipulation {
+    format = '[[\?color=#f00&show WINDOW:] {format_window}][\?soft  ]'
+    format += '[[\?color=#f00&show SCRATCHPAD:] {format_scratchpad}]'
+    format += '[\?color=#f00&show DIVIDER] '
+    format += '[\?color=#0ff&show NAME:] {name} '
+    format += '[\?color=#0ff&show CLASS:] {class} '
+    format += '[\?color=#0ff&show FLOATING:] {floating} '
+    format += '[\?color=#0ff&show SCRATCHPAD:] {scratchpad} '
+    format += '[\?color=#0ff&show URGENT:] {urgent} '
+    format += '[\?color=#0ff&show URGENT_SP:] {urgent_sp} '
+    format += '[\?color=#0ff&show WINDOW:] {window}'
+    format += '[\?color=#0ff&show ID:] {id} '
+
+    format_window = '\?max_length=30 [\?if=is_urgent&color=urgent \u2756'
+    format_window += '|[\?if=is_focused&color=focused \u2756'
+    format_window += '|[\?if=is_scratchpad&color=scratchpad \u2756'
+    format_window += '|[\?if=is_floating&color=floating \u2756'
+    format_window += '|\?color=unfocused \u2756]]]] {name}'
+
+    format_scratchpad = '[\?color=#ff0&show INDEX:] {index} '
+    format_scratchpad += '[\?color=#ff0&show NAME:] {name} '
+    format_scratchpad += '[\?color=#ff0&show CLASS:] {class} '
+    format_scratchpad += '[\?color=#ff0&show ID:] {id} '
+}
+
+# keep it simple, stupid
+window_manipulation {
+    format_window = '\?max_length=20 [\?if=is_focused&color=good \u2756'
+    format_window += '|\?color=bad \u2756] {name}'
+}
+
+# add your magic snippets here
+window_manipulation {
+    format = '???'
+}
+```
+
+SAMPLE OUTPUT
+{'full_text': u'window_manipulation'}
+
+complicated
+{'full_text': u'(too complicated)'}
+"""
+
+from threading import Thread
+
+import i3ipc
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    button_focus = None
+    button_next = None
+    button_previous = None
+    format = '{format_window}'
+    format_scratchpad = None
+    format_separator = ' '
+    format_window = None
+    urgent_type = None
+
+    def post_config_hook(self):
+        if self.format_window is None:
+            self.format_window = ''
+        if self.format_scratchpad is None:
+            self.format_scratchpad = ''
+
+        self.fmt_scratchpad = self.fmt_window = self.id = \
+            self.window_name = self.class_name = self.which_data = None
+        self.count_floating = self.count_scratchpad = \
+            self.count_urgent_sp = self.count_urgent = self.count_window = 0
```

Comment:
can you not do this I am stopping my review here because this is not readable.

please do not do this multiple assignment like this because it is horrible.  Also try to avoid using `\` to extend lines as it usually means the code is badly written